src/components/modebar/buttons.js

```diff
@@ -533,8 +533,19 @@ function handleGeo(gd, ev) {
 
         if(attr === 'zoom') {
             var scale = geoLayout.projection.scale;
+            var minscale = geoLayout.projection.minscale;
+            var maxscale = geoLayout.projection.maxscale;
+
+            if(maxscale === -1) maxscale = Infinity;
```

Comment:
Let's add
```js
var oldScale = scale;
```
here.
Then we should only call `Registry.call('_guiRelayout', ...` below only when `newScale !== oldScale`.

Reply:
Good idea! I used the existing `scale` variable for the check: d69d7994e6ad9ee86a2d4a30e6c5b8dde06395f4